src/Symfony/Component/HttpClient/HttpClient.php

```diff
@@ -35,6 +36,8 @@ public static function create(array $defaultOptions = [], int $maxHostConnection
             }
 
             @trigger_error('Configure the "curl.cainfo", "openssl.cafile" or "openssl.capath" php.ini setting to enable the CurlHttpClient', E_USER_WARNING);
+        } elseif (class_exists(ConnectionLimitingPool::class)) {
+            return new AmpHttpClient($defaultOptions, null, $maxHostConnections, $maxPendingPushes);
```

Comment:
In a future version, this would probably also be the case if `amphp/http-client` is installed, but Curl does not support HTTP/2?

Reply:
we can do it now! updated

Reply:
is `AmpHttpClient` faster or slower than a `CurlHttpClient` without HTTP/2 support ?

Reply:
it mostly makes no diff, perf-wise - in HTTP/1.1, the network is the limiting factory
but yes, generally speaking, curl implements more logic in C.

Reply:
@stof: For very large downloads on high-speed connections it will probably be a bit slower, because the flow control isn't as good as the flow control of the Linux kernel for TCP. However, that's probably not the most used case. If you have multiple concurrent requests, HTTP/2 is probably a win.

Reply:
`tcp_nodelay` might another reason: by default, it is enabled when using curl, because curl already aggregates writes. I don't know if that's the case for amp - but `tcp_nodelay` is disabled when using it by default.

Reply:
I thought `tcp_nodelay` would be enabled, but it seems it isn't. But it can simply be done by passing a `ConnectContext` to [`DefaultConnectionFactory`](https://github.com/amphp/http-client/blob/8395dc643b9f2e740231e9ca579eac0ba96b999e/src/Connection/DefaultConnectionFactory.php#L31).

Reply:
`tcp_nodelay` added